serverlessworkflow/modules/ROOT/pages/getting-started/cncf-serverless-workflow-specification-support.adoc

```diff
@@ -1 +1,368 @@
+
+.Example of a Java function implementation
+[source,java]
+----
+public class MyInterfaceOrClass {
+
+    public void myMethod(JsonNode workflowData) {
+        // do whatever I want with the JsonNode:
+        // { "workflowdata": {} }
+    }
+    
+    public JsonNode myMethod(JsonNode workflowData) {
+        // do whatever I want with the JsonNode:
+        // { "workflowdata": {} }
+        // return the modified content:
+        return workflowData;
+    }
+}
+----
+
+To call this function within your workflow, you can extract the JSON value you need via a xref:core/understanding-jq-expressions.adoc[`jq` expression] or pass it without any arguments. In this case, the caller sends the whole payload. For example:
+
+.Example of a reference to a Java function in the Operation state
+[source,json]
+----
+{
+  "states": [
+    {
+      "name": "myState",
+      "type": "operation",
+      "actions": [
+        {
+          "name": "callJavaFunctionAction",
+          "functionRef": {
+            "refName": "myFunction"
+          }
+        }
+      ]
+    }
+  ]
+}
+----
+
+Alternatively, you can only pass the necessary data:
+
+.Example of a reference to a Java function in the Operation state passing manipulated data
+[source,json]
+----
+{
+  "states": [
+    {
+      "name": "myState",
+      "type": "operation",
+      "actions": [
+        {
+          "name": "callJavaFunctionAction",
+          "functionRef": {
+            "refName": "myFunction",
+            "arguments": {
+              "data": "${ .my.expression.to.data }"
+            }
+          }
+        }
+      ]
+    }
+  ]
+}
+----
+
+[#events]
+=== Workflow Model - Events
+
+Kogito supports the capabilities of the event according to the {spec_doc_url}#Event-Definition[specification defition of an event]. However, event `correlation` has limited support as per specification definition.
+
+For more details about the event correlation feature that Kogito supports, see the guide xref:eventing/event-correlation-with-workflows.adoc[Event Correlation with Workflows].
+
+[#workflow_data]
+=== Workflow Data
+
+Kogito has the full implementation of the data manipulation feature via either `jq` (default) or `jsonpath`.
+
+JSON Path can have some limitations in specific scenarios, mainly when the use case requires complex data manipulation. See the {spec_doc_url}#workflow-data[Workflow Data] chapter in the specification.
+
+Kogito also supports xref:core/state-data-filtering[data filtering]. For more information, check the section {spec_doc_url}#state-data-filters[State Data Filters] in the specification documentation.
+
+[#expressions]
+=== Expressions
+
+Kogito supports either `jq` or `jsonpath` to define workflow expressions. As the {spec_doc_url}#workflow-expressions[specification defines], `jq` is the default expression language. If you want to use `jsonpath` instead, set the attribute `expressionLang` in the workflow definition:
```

Comment:
maybe rephrase as it is stated on line 304 as well?

Reply:
I think the other sentence will be in the new guide